KubeArmor/core/hookHandler.go

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package core
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+	kl "github.com/kubearmor/KubeArmor/KubeArmor/common"
+	cfg "github.com/kubearmor/KubeArmor/KubeArmor/config"
+	"github.com/kubearmor/KubeArmor/KubeArmor/state"
+	tp "github.com/kubearmor/KubeArmor/KubeArmor/types"
+)
+
+func (dm *KubeArmorDaemon) HandleFile(file string) {
+	f, err := os.Open(file)
+	if err != nil {
+		log.Println("Error opening file:", err)
+	}
+
+	decoder := json.NewDecoder(f)
+	for {
+		var containerData tp.Container
+
+		err = decoder.Decode(&containerData)
+		if err != nil {
+			dm.Logger.Warnf("Reached EOF")
+		}
+		dm.handleContainerCreate(containerData)
+		if err == io.EOF {
+			// End of file reached
+			break
+		}
+	}
+	f.Close()
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatal("Error creating new watcher:", err)
+	}
+	defer w.Close()
+
+	err = w.Add(file)
+	if err != nil {
+		log.Fatal("Error adding file to watcher:", err)
+	}
+
+	for {
+		select {
+		case err, ok := <-w.Errors:
+			if !ok {
+				dm.Logger.Warnf("Returning 1")
+				return
+			}
+			log.Println("Watcher error:", err)
+
+		case e, ok := <-w.Events:
+			if !ok {
+				dm.Logger.Warnf("Returning 2")
+				return
+			}
+
+			if e.Op&fsnotify.Write == fsnotify.Write {
+				dm.Logger.Warnf("Detected changes in output.json")
+				f, err := os.Open(file)
+				if err != nil {
```

Comment:
Let's use the sockets approach, we opted for fsnotify only for testing and quick turnaround

Reply:
Issue with handling containers created before kubearmor starts.